modules/core/src/lib/layer.ts

```diff
@@ -434,11 +435,39 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
 
   // Default implementation
   // Sublayers chould override this method to provide an accurate calculation of the bounds
-  getBounds(): [number[], number[]] | null {
+  getBounds(attributes?: string[]): [number[], number[]] | null {
+    return this._getBounds(attributes || ['positions', 'instancePositions']);
+  }
+
+  private _getBounds(attributes: string[]): [number[], number[]] | null {
     const attributeManager = this.getAttributeManager();
-    if (!attributeManager) return null;
-    const {positions, instancePositions} = attributeManager.attributes;
-    return (positions || instancePositions)?.getBounds();
+    if (!attributeManager || this.internalState === null) return null;
+    const {cachedBounds} = this.internalState;
+
+    // Detect if bounds of any requested attribute have changed
+    let boundsChanged = false;
+    for (const a of attributes) {
```

Comment:
Convention is to always type out variables:
```suggestion
    for (const attribute of attributes) {
```
However since we are dealing both with attributes names and actual attribute objects, I recommend
```suggestion
    for (const attributeName of attributeNames) {
```